src/Data.Json.JTable.Internal.purs

```diff
@@ -0,0 +1,214 @@
+module Data.Json.JTable.Internal
+  ( renderJTableRaw, renderRows, renderThead, renderTbody, tsToRows, sortTree
+  , Tree(..), tPath, tWidth, tHeight, tKids
+  , Cell(..), cCursor, cWidth, cHeight, cJsonPrim
+  , JPath(..), Table(..)
+  , tFromJson, tMergeArray, widthOfPrimTuple, insertHeaderCells
+  , cFromJson, cMergeObj, mergeObjTuple
+  , _cN, toPrim, strcmp, _nattr, _cspan, _rspan
+  ) where
+
+import Math (max)
+import Data.Either
+import Data.Maybe (Maybe(..), fromMaybe, isJust)
+import Data.Tuple
+import Data.String (joinWith, localeCompare)
+import Data.Array
+import Data.Foldable
+import qualified Data.Array.Unsafe as AU
+import qualified Data.StrMap as M
+import Data.Foldable (foldl, any, all, mconcat, elem)
+import Data.Traversable (for)
+import Control.Alt
+import Control.Apply
+import Control.Bind
+import Control.Monad
+import Control.MonadPlus
+
+import Data.Argonaut.Core
+import Data.Argonaut.JCursor
+import Data.Argonaut.Parser (jsonParser)
+import Text.Smolder.HTML (thead, tbody, tr, th, td)
+import Text.Smolder.Markup (Markup(..), Attribute(..), attribute, (!))
+
+-- path of object keys, with array indices omitted
+type JPath = [String]
+
+-- rows of cells
+type Table = [[Cell]]
```

Comment:
`cFromJson` recursively merges lists of tables. I guess if we rewrite that to work in terms of Markup, it could work, but I'm not sure we'd gain much in terms of storage.